### PR TITLE
Chore/vrf 526 add ctf test vrfv2plus

### PR DESCRIPTION
**Pull request by iljapavlovs**

### CODE REVIEW

File: integration-tests/actions/vrfv2plus/vrfv2plus_steps.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+	"github.com/smartcontractkit/chainlink/integration-tests/client"
+	"github.com/smartcontractkit/chainlink/integration-tests/contracts"
+	chainlinkutils "github.com/smartcontractkit/chainlink/v2/core/utils"
+	"math/big"
+)
+
+var (
+	ErrNodePrimaryKey         = "error getting node's primary ETH key"
+	ErrCreatingProvingKeyHash = "error creating a keyHash from the proving key"
+	ErrCreatingProvingKey     = "error creating a keyHash from the proving key"
```
Comment (jinhoonbang) [inline integration-tests/actions/vrfv2plus/vrfv2plus_steps.go:19]: Duplicate?

File: integration-tests/actions/vrfv2plus/vrfv2plus_steps.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+	ErrRegisterProvingKey     = "error registering proving keys"
+	ErrEncodingProvingKey     = "error encoding proving key"
+	ErrCreatingVRFv2Key       = "error creating VRFv2 key"
```
Comment (jinhoonbang) [inline integration-tests/actions/vrfv2plus/vrfv2plus_steps.go:22]:
```suggestion
	ErrCreatingVRFv2PlusKey       = "error creating VRFv2Plus key"
```

File: integration-tests/actions/vrfv2plus/vrfv2plus_steps.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	ErrDeployBlockHashStore   = "error deploying blockhash store"
+	ErrDeployCoordinator      = "error deploying VRFv2 CoordinatorV2"
```
Comment (jinhoonbang) [inline integration-tests/actions/vrfv2plus/vrfv2plus_steps.go:24]:
```suggestion
	ErrDeployCoordinator      = "error deploying VRF CoordinatorV2Plus"
```

File: integration-tests/actions/vrfv2plus/vrfv2plus_steps.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+	ErrAdvancedConsumer       = "error deploying VRFv2 Advanced Consumer"
+	ErrABIEncodingFunding     = "error Abi encoding subscriptionID"
+	ErrSendingLinkToken       = "error sending Link token"
+	ErrCreatingVRFv2Job       = "error creating VRFv2 job"
```
Comment (jinhoonbang) [inline integration-tests/actions/vrfv2plus/vrfv2plus_steps.go:28]:
```suggestion
	ErrCreatingVRFv2PlusJob       = "error creating VRFv2Plus job"
```

File: integration-tests/actions/vrfv2plus/vrfv2plus_steps.go [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+var (
+	ErrNodePrimaryKey         = "error getting node's primary ETH key"
+	ErrCreatingProvingKeyHash = "error creating a keyHash from the proving key"
+	ErrCreatingProvingKey     = "error creating a keyHash from the proving key"
+	ErrRegisterProvingKey     = "error registering proving keys"
+	ErrEncodingProvingKey     = "error encoding proving key"
+	ErrCreatingVRFv2Key       = "error creating VRFv2 key"
+	ErrDeployBlockHashStore   = "error deploying blockhash store"
+	ErrDeployCoordinator      = "error deploying VRFv2 CoordinatorV2"
+	ErrAdvancedConsumer       = "error deploying VRFv2 Advanced Consumer"
```
Comment (jinhoonbang) [inline integration-tests/actions/vrfv2plus/vrfv2plus_steps.go:25]:
```suggestion
	ErrAdvancedConsumer       = "error deploying VRFv2Plus Advanced Consumer"
```

File: integration-tests/contracts/ethereum_vrfv2plus_contracts.go [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+	"math/big"
+)
+
+type EthereumVRFCoordinatorV2Plus struct {
+	address     *common.Address
+	client      blockchain.EVMClient
+	coordinator *vrf_coordinator_v2plus.VRFCoordinatorV2Plus
+}
+
+// EthereumVRFv2LoadTestConsumer represents VRFv2 consumer contract for performing Load Tests
```
Comment (jinhoonbang) [inline integration-tests/contracts/ethereum_vrfv2plus_contracts.go:21]:
```suggestion
// EthereumVRFv2PlusLoadTestConsumer represents VRFv2Plus consumer contract for performing Load Tests
```

File: integration-tests/contracts/ethereum_vrfv2plus_contracts.go [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+type EthereumVRFv2PlusLoadTestConsumer struct {
+	address  *common.Address
+	client   blockchain.EVMClient
+	consumer *vrf_v2plus_load_test_with_metrics.VRFV2PlusLoadTestWithMetrics
+}
+
+// DeployVRFCoordinatorV2 deploys VRFV2 coordinator contract
```
Comment (jinhoonbang) [inline integration-tests/contracts/ethereum_vrfv2plus_contracts.go:28]:
```suggestion
// DeployVRFCoordinatorV2Plus deploys VRFV2Plus coordinator contract
```

File: integration-tests/contracts/ethereum_vrfv2plus_contracts.go [hunk 8]
```diff
@@ -0,0 +1,245 @@
...
+
+func (v *EthereumVRFv2PlusLoadTestConsumer) Address() string {
+	return v.address.Hex()
+}
+func (v *EthereumVRFv2PlusLoadTestConsumer) RequestRandomness(keyHash [32]byte, subID *big.Int, requestConfirmations uint16, callbackGasLimit uint32, nativePayment bool, numWords uint32, requestCount uint16) error {
+	opts, err := v.client.TransactionOpts(v.client.GetDefaultWallet())
+	if err != nil {
+		return err
+	}
+	//opts *bind.TransactOpts, _subId *big.Int, _requestConfirmations uint16, _keyHash [32]byte, _callbackGasLimit uint32, _nativePayment bool, _numWords uint32, _requestCount uint16
```
Comment (jinhoonbang) [inline integration-tests/contracts/ethereum_vrfv2plus_contracts.go:178]: *nit remove?

File: integration-tests/contracts/ethereum_vrfv2plus_contracts.go [hunk 9]
```diff
@@ -0,0 +1,245 @@
...
+	})
+}
+
+func (v *EthereumVRFv2PlusLoadTestConsumer) GetLoadTestMetrics(ctx context.Context) (*VRFLoadTestMetrics, error) {
+	requestCount, err := v.consumer.SRequestCount(&bind.CallOpts{
+		From:    common.HexToAddress(v.client.GetDefaultWallet().Address()),
+		Context: ctx,
+	})
+	if err != nil {
+		return &VRFLoadTestMetrics{}, err
```
Comment (jinhoonbang) [inline integration-tests/contracts/ethereum_vrfv2plus_contracts.go:205]: can do `return nil, err`

File: integration-tests/smoke/vrfv2plus_test.go [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+		vrfv2plus_constants.MinimumConfirmations,
+		vrfv2plus_constants.CallbackGasLimit,
+		false,
+		vrfv2plus_constants.NumberOfWords,
+		vrfv2plus_constants.RandomnessRequestCountPerRequest,
+	)
+	require.NoError(t, err)
+
+	gom := gomega.NewGomegaWithT(t)
+	timeout := time.Minute * 1
```
Comment (jinhoonbang) [inline integration-tests/smoke/vrfv2plus_test.go:98]: do we need to hard code timeout here or can we use `testutils.WaitTimeout(t)`

File: .github/workflows/integration-tests.yml [hunk 11]
```diff
@@ -273,6 +273,10 @@ jobs:
             nodes: 1
             os: ubuntu20.04-8cores-32GB
             pyroscope_env: ci-smoke-vrf2-evm-simulated
+          - name: vrfv2plus
+            nodes: 1
+            os: ubuntu20.04-8cores-32GB
```
Comment (skudasov) [inline .github/workflows/integration-tests.yml:278]: Consider using `ubuntu-latest` since it's just a smoke test with 1 CL Node.

**Review by skudasov — APPROVED** [review]

**Pull request merged**